src/api.js

```diff
@@ -1678,6 +1684,26 @@ var resolveTask = async function(req, res, taskId, runId, target) {
     return;
   }
 
+  // Ensure that all blob artifacts which were created are present before
+  // allowing resolution as 'completed'
+  if (target === 'completed') {
+    let haveAllBlobs = true;
+    await this.Artifact.query({
+      taskId,
+      runId,
+      storageType: 'blob',
+      present: false,
+    }, {
+      limit: 1,
+      handler: () => { haveAllBlobs = false; },
+    });
+
+    if (!haveAllBlobs) {
+      return res.reportError('InputError',
```

Comment:
Or  `RequestConflict` as it's sort of the request that conflicts with the state...

Reply:
btw, thanks for adding this... It's a huge safety feature... and means we can trust the workers a little bit less :)

Reply:
no worries.